### ISSUE TITLE
1.7 JSX template error

### ISSUE THREAD

**Issue by thetarnav** [solidjs/solid#1663]
### Describe the bug

The new shorter template strings seem to have some edge cases to polish.
Here is one that [worked in 1.6](https://solid-playground.netlify.app/?version=1.6.2#NobwRAdghgtgpmAXGGUCWEwBowBcCeADgsrgM4Ae2YZA9gK4BOAxiWGjIbY7gAQi9GcCABM4jXgF9eAM0a0YvADo1aAGzQiAtACsyAegDucAEYqA3EohXmtCGT4yIvALy8AFAEpXAPl4Byf0trCFt7PgBhO1xhPjcvX34rXkE4XCZnd2SU3gAeETQANx9snLyyQigIEucynNyoHw0IAGtc-UbS+v0Kqpq68srqmIpcdt7qrpTciZ8QJy9JcaH+7oLi7M9gyWCrIVFxdwSXP1yoiBiL3n0fLF4RWmZ6eAuAOgBzNIBRNTgX3AAQvgAJIiLJgKCEQgqTwAQi2VjAkgAukA)
But [doesn't work in 1.7](https://stackblitz.com/edit/solid-1-7-template-bug?file=src%2FApp.tsx)

```ts
<div>
  <span>
    <a>link</a>
  </span>
  <span>text</span>
  <span>{fn()}</span>
</div>
```

in 1.6 it compiled to something like this:
<details>
<summary>Code snippet</summary>

```ts
const _tmpl$ = /*#__PURE__*/template(`<div><span><a>link</a></span><span>text</span><span></span></div>`, 10);

const fn = () => '';

const Content = () => {
  return (() => {
    const _el$ = _tmpl$.cloneNode(true),
          _el$2 = _el$.firstChild,
          _el$3 = _el$2.nextSibling,
          _el$4 = _el$3.nextSibling;

    insert(_el$4, fn);

    return _el$;
  })();
};
```

</details>

When in 1.7 it compiles like this:

<details>
<summary>Code snippet</summary>

```ts
const _tmpl$ = /*#__PURE__*/ _$template(`<div><span><a>link</span><span>text</span><span>`);
const fn = ()=>'';
const Content = ()=>{
    return (()=>{
        const _el$ = _tmpl$()
          , _el$2 = _el$.firstChild
          , _el$3 = _el$2.nextSibling
          , _el$4 = _el$3.nextSibling;
        _$insert(_el$4, fn);
        return _el$;
    }
    )();
};
```

</details>

Here is the error
```
Uncaught TypeError: Cannot set properties of null (setting 'textContent')
    at insertExpression (chunk-Z7W7ZHXW.js?v=12db9d9c:553:38)
    at Object.fn (chunk-Z7W7ZHXW.js?v=12db9d9c:336:35)
    at runComputation (chunk-ANANZPPK.js?v=12db9d9c:752:22)
    at updateComputation (chunk-ANANZPPK.js?v=12db9d9c:735:3)
    at createRenderEffect (chunk-ANANZPPK.js?v=12db9d9c:229:5)
    at insert (chunk-Z7W7ZHXW.js?v=12db9d9c:336:3)
    at App.tsx:11:5
    at Content (App.tsx:13:5)
    at chunk-ANANZPPK.js?v=12db9d9c:588:12
    at untrack (chunk-ANANZPPK.js?v=12db9d9c:460:12)
```

### Your Example Website or App

https://stackblitz.com/edit/solid-1-7-template-bug?file=src%2FApp.tsx

**Comment by ryansolid** [comment]
Thanks I see the anchor isn't closing correctly.

**Comment by ryansolid** [comment]
Yeah it's a weird one. Feels like maybe browser bug if weren't consistent in Firefox and Chrome. It seems like if you don't close the anchor tag it will close it in the right place, but assume that any more sibling inline elements should be in their own anchors. Knowing this I'm just going to close all anchor tags and move on.

**Comment by mattvj** [comment]
It isn't just anchor tags that are having issues.

```
<div class="flex flex-col gap-4">
      <p class="text-sm">
        <span>This is some text for link </span>
        <b class="cursor-pointer text-blue-400 underline" href="somelink">
          My Link
        </b>
      </p>
      <p class="text-sm">
        This is a second paragraph
      </p>
      <p class="text-sm">
        Third paragraph
      </p>
    </div>
```
This is outputing: 

`<div class="flex flex-col gap-4"><p class="text-sm"><span>This is some text for link </span><b class="cursor-pointer text-blue-400 underline" href="somelink">My Link</p><p class="text-sm">This is a second paragraph</p><p class="text-sm">Third paragraph`

The bold is finding its way into the second and third paragraphs.

[playground](https://playground.solidjs.com/anonymous/fec7d77d-3f19-44b6-a980-7f9db2e1dcc9)

**Comment by ryansolid** [comment]
Thanks for reporting.. so bold looks ahead for text nodes it looks like. I wonder what other elements are like this. Luckily the fix is simple as long as we can identify them all.

**Comment by ryansolid** [comment]
Ok should be fixed in `babel-plugin-jsx-dom-expressions` 0.36.2.

I haven't done a Solid release yet, but blowing away your lockfile and npm modules should get you the latest

**Comment by robertwayne** [comment]
It looks like labels are subject to this as well, FYI.
